emmet-core/emmet/core/qchem/calculation.py

```diff
@@ -0,0 +1,437 @@
+            if qcinput.molecule.as_dict()
+            else None,
+            rem=qcinput.rem,
+            job_type=qcinput.rem.get("job_type", None),
+            opt=qcinput.opt,
+            pcm=qcinput.pcm,
+            solvent=qcinput.solvent,
+            smx=qcinput.smx,
+            vdw_mode=qcinput.vdw_mode,
+            scan_variables=qcinput.scan,
+            van_der_waals=qcinput.van_der_waals,
+        )
+
+
+class CalculationOutput(BaseModel):
+    """Document defining QChem calculation outputs."""
+
+    optimized_molecule: Dict[str, Any] = Field(
+        None,
+        description="optimized geometry of the molecule after calculation in case of opt, optimization or ts",
+    )
+
+    mulliken: Dict[str, Any] = Field(
+        None, description="Calculate Mulliken charges on the atoms"
+    )
+
+    esp: Dict[str, Any] = Field(
+        None,
+        description="Calculated charges on the atoms if esp calculation has been performed",
+    )
+
+    resp: Dict[str, Any] = Field(
+        None,
+        description="Calculated charges on the atoms if resp calculation has been performed",
+    )
+
+    nbo_data: Dict[str, Any] = Field(
+        None, description="NBO data if analysis has been performed."
+    )
+
+    frequencies: Dict[str, Any] = Field(
+        None,
+        description="Calculated frequency modes if the job type is freq or frequency",
+    )
+
+    frequency_modes: Union(List, str) = Field(
+        None, description="The list of calculated frequency mode vectors"
+    )
+
+    final_energy: Union(str, float) = Field(
+        None,
+        description="The final energy of the molecule after the calculation is complete",
+    )
+
+    enthalpy: Union(str, float) = Field(
+        None,
+        description="The total enthalpy correction if a frequency calculation has been performed",
+    )
+
+    entropy: Union(str, float) = Field(
+        None,
+        description="The total entropy of the system if a frequency calculation has been performed",
+    )
+
+    scan_energies: Dict[str, Any] = Field(
+        None,
+        description="A dictionary of the scan energies with their respective parameters",
+    )
+
+    scan_geometries: Dict[str, Any] = Field(
+        None, description="optimized geometry of the molecules after the geometric scan"
+    )
+
+    scan_molecules: Dict[str, Any] = Field(
+        None,
+        description="The constructed pymatgen molecules from the optimized scan geometries",
+    )
+
+    pcm_gradients: Union(Dict[str, Any], np.ndarray) = Field(
+        None,
+        description="The parsed total gradients after adding the PCM contributions.",
+    )
+
+    cds_gradients: Union(Dict[str, Any], np.ndarray) = Field(
+        None, description="The parsed CDS gradients."
+    )
+
+    dipoles: Dict[str, Any] = Field(
+        None, description="The associated dipoles for Mulliken/RESP/ESP charges"
+    )
+
+    gap_info: Dict[str, Any] = Field(
```

Comment:
I remember saying at one point that I had no interest in including HOMO-LUMO in MPcules,  but that if someone else wanted to, I wouldn't stop them. Now you've started the work, and I feel myself tempted to go back on my work and fight you.

But I won't. I will stay strong, even though the HOMO-LUMO beast is evil.